ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixParticipant.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.VerifiableProperties;
+import org.apache.helix.HelixManager;
+import org.apache.helix.HelixManagerFactory;
+import org.apache.helix.InstanceType;
+import org.apache.helix.model.LeaderStandbySMD;
+import org.apache.helix.participant.StateMachineEngine;
+
+
+public class HelixParticipant implements ClusterParticipant {
+  private final VerifiableProperties properties;
+  private HelixManager manager;
+  private final String clusterName;
+  private final String zkConnectStr;
+
+  HelixParticipant(VerifiableProperties properties, ClusterMapConfig clusterMapConfig) {
+    this.properties = properties;
+    clusterName = clusterMapConfig.clusterMapClusterName;
+    if (clusterName.isEmpty()) {
+      throw new IllegalStateException("Clustername is empty in clusterMapConfig");
+    }
+    zkConnectStr = clusterMapConfig.clusterMapParticipantZkConnectString;
+  }
+
+  @Override
+  public void initialize(String hostName, int port) throws Exception {
+    String instanceName = hostName + "_" + port;
+    manager = HelixManagerFactory.getZKHelixManager(clusterName, instanceName, InstanceType.PARTICIPANT, zkConnectStr);
+    StateMachineEngine stateMachineEngine = manager.getStateMachineEngine();
+    stateMachineEngine.registerStateModelFactory(LeaderStandbySMD.name, new AmbryStateModelFactory(zkConnectStr));
+    manager.connect();
+  }
+
+  @Override
+  public void terminate() {
+    manager.disconnect();
```

Comment:
It does not throw any checked exceptions. Even if it fails, it should be fine as this is in the exit path.